Mods/DebugMod/Src/DebugMod.cpp

```diff
@@ -42,10 +42,20 @@
 
 DebugMod::~DebugMod()
 {
+	if (m_TrackCamActive)
+	{
+		DisableTrackCam();
+	}
+
     const ZMemberDelegate<DebugMod, void(const SGameUpdateEvent&)> s_Delegate(this, &DebugMod::OnFrameUpdate);
-    Globals::GameLoopManager->UnregisterFrameUpdate(s_Delegate, 1, EUpdateMode::eUpdatePlayMode);
 
-    if (m_TrackCamActive) DisableTrackCam();
+	try
+	{
+		Globals::GameLoopManager->UnregisterFrameUpdate(s_Delegate, 1, EUpdateMode::eUpdatePlayMode);
+	} catch (std::exception e)
+	{
+		Logger::Error("Exception: {}", e.what());
+	}
```

Comment:
I believe the game doesn't generally throw C++ exceptions, especially from functions like these. I've only seen `std::bad_alloc` exceptions thrown during memory allocation failures. Did you get any exceptions handled here?

Reply:
It's probably overkill on my part - it was getting stuck in the unregisterFrameUpdated, now and then when unloading the mod. The other changes may have fixed it since.

I'll remove it locally, see how it is and then update this PR if things seem stable enough - otherwise, I'll investigate a bit more if the issues start happening again